lib/any_login/engine.rb

```diff
@@ -23,29 +20,29 @@ class Engine < ::Rails::Engine
     private
 
     def load_provider
-      case AnyLogin.provider || provider
-      when :authlogic
-        require 'any_login/providers/authlogic'
-      when :devise
-        require 'any_login/providers/devise'
-      when :clearance
-        require 'any_login/providers/clearance'
-      when :sorcery
-        require 'any_login/providers/sorcery'
+      if (selected_provider = AnyLogin.provider || provider)
+        provider_with_module = "AnyLogin::Provider::#{selected_provider.to_s.capitalize}"
+        if Object.const_defined?(provider_with_module)
+          AnyLogin.provider = provider_with_module
+        else
+          AnyLogin.provider = selected_provider.to_s
+        end
```

Comment:
I tried to keep the external interface the same for users here, the main difference is this is stored during runtime as a string instead of a class. With custom providers, trying to store this as a class doesn't work since the class isn't accessible until after load. This is the same method Devise uses for the parent mailer class, storing it as a string likely for the same reason: https://github.com/heartcombo/devise/search?q=parent_mailer